new-new-packages/compiler/src/discover/run.rs

```diff
@@ -17,160 +17,164 @@ use super::{
+            environment = run_body(db, &body, environment);
+            environment.get(body.out_id()).transitive()
         }
         Expression::Call {
             function,
             arguments,
-        } => run_call(db, input, function, arguments, environment.clone()),
+        } => run_call(db, function, arguments, environment.clone()),
         Expression::Error => DiscoverResult::ErrorInHir,
     };
     environment.store(id, result);
     environment
 }
 pub(super) fn run_call(
     db: &dyn Discover,
-    input: Input,
     function: hir::Id,
     arguments: Vec<hir::Id>,
     environment: Environment,
 ) -> DiscoverResult {
-    if let &[builtin_function_index] = &function.0[..] {
+    if let &[builtin_function_index] = &function.local[..] {
         if let Some(builtin_function) = builtin_functions::VALUES.get(builtin_function_index) {
             return run_builtin_function(
                 db,
-                input,
                 builtin_function.to_owned(),
                 arguments.to_owned(),
                 environment,
             );
         }
     }
 
-    let lambda = match environment.get(&function).unwrap().transitive()? {
+    let lambda = match environment.get(&function).transitive()? {
         Value::Lambda(lambda) => lambda,
         it => panic!("Tried to call something that wasn't a lambda: {:?}", it),
     };
-    let lambda_hir = match db.find_expression(input.to_owned(), lambda.id)? {
+    let lambda_hir = match db.find_expression(lambda.id)? {
         Expression::Lambda(lambda) => lambda,
         hir => panic!(
             "Discover lambda is not backed by a HIR lambda, but `{}`.",
             hir
         ),
     };
 
-    let lambda_parent = if let Some(lambda_parent_id) = function.parent() {
-        match db.find_expression(input.to_owned(), lambda_parent_id)? {
-            Expression::Body(body) => body,
-            hir => panic!("A called lambda's parent isn't a body, but `{}`", hir),
-        }
-    } else {
-        let (hir, _) = db.hir(input.to_owned()).unwrap();
-        hir.as_ref().to_owned()
-    };
-    let function_name = lambda_parent.identifiers.get(&function).unwrap();
-    log::trace!("Calling function `{}`", &function_name);
+    // let lambda_parent = if let Some(lambda_parent_id) = function.parent() {
+    //     match db.find_expression(lambda_parent_id)? {
+    //         Expression::Body(body) => body,
+    //         hir => panic!("A called lambda's parent isn't a body, but `{}`", hir),
+    //     }
+    // } else {
+    //     let (hir, _) = db.hir(function.input.to_owned()).unwrap();
+    //     hir.as_ref().to_owned()
+    // };
+    // match lambda_parent.identifiers.get(&function) {
+    //     Some(function_name) => log::trace!("Calling function `{}`", &function_name),
+    //     // TODO: Inline lambdas don't have names, but for some other lambdas the name is not found.
+    //     None => log::trace!("Calling function {}", function),
+    // };
```

Comment:
What's up with this?

Reply:
This was logging for every single function call, which quickly adds up when a whole file is invalidated at once, and, therefore, also seemingly slowed the VS Code Extension down a bit. I'm not sure how we want to proceed with this in the future – debug logging of the called functions might still be useful, especially when running the compiler from the console, but in VS Code, it's too much. Maybe conditionally run this only from the CLI?